### PR TITLE
HPCC-23901 add EFS subchart and configurations

### PR DESCRIPTION
**Pull request by xwang2713**
Following changed are made:
1) Setup a EFS
2) Add a EFS subchart under hpcc/charts/
3) Add a dependency in hpcc/Chart.yaml
4) Modify values.schema.json values.yaml for EFS entries
5) Modify templates/dali.yaml, datastorage-pvc.yaml anddllserver-pvc.yaml for EFS

What user need to do are:
1) setup EFS
2) Add EFS settings in charts/efs/values.yaml
3) set global.efs.enabled to true and set storageClass to "efs" in hpcc/values.yaml

I did write a initial blog to show how to setup EFS/EKS on AWS to deploy HPCC Systems Platform cluster

 Signed-off-by: Xiaoming Wang <xiaoming.wang@lexisnexis.com>
@jakesmith please review

@JamesDeFabia I will send the initial blog in *.md format. You can help to review it.

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:

Tested on AWS region us-east-2 for both "efs" and  "gp2" storage class. Did test scale with "efs"

### CODE REVIEW

File: dockerfiles/hpcc/charts/efs/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+apiVersion: v2
+name: efs
+version: 0.1.0
+appVersion: 0.1.0
+description:
+#sources:
+#- https://github.com/hpcc-systems/HPCC-Platform/dockerfiels/hpcc/charts/efs
```
Comment (JamesDeFabia) [inline dockerfiles/hpcc/charts/efs/Chart.yaml:7]: dockerfiels?

**Comment by JamesDeFabia** [comment]
@xwang2713 I returned the edited MD file to you. Let me know if you need anything else. 
@jakesmith can you review the PR?

**Comment by xwang2713** [comment]
Since Jake make nfs working in Google Cloud as existing PersistentVolumeClaim I will do the same. So no need to change <HPCC Platform>/dockerfiles/hpcc.  I create a HELM Char for EFS and will ask student (Google Cloud) to create a similar one based on Jake's work. We probable will host these charts in HPCC-Kube project.  I will update documentation.

**Comment by jakesmith** [comment]
I'm trying to release that as an example at the moment, hopefully today.
We are also going to release Azure examples for storage setup.

We think in all cases they should be separate helm charts, not integrated into our hpcc charts, but installed independently.
And the hpcc cluster (as it can be now) to be configured to point to existing PVC's.

Are you able to restructure this EFS example, such that it creates 3 PVC's for HPCC as an example helm chart?

**Comment by xwang2713** [comment]
Yes, I already did.

**Comment by xwang2713** [comment]
If not use HPCC-Kube git hub project do we have a web server can host charts?

**Comment by richardkchapman** [comment]
We host the hpcc chart on github

**Comment by richardkchapman** [comment]
https://github.com/hpcc-systems/helm-chart

**Comment by xwang2713** [comment]
OK

**Comment by xwang2713** [comment]
Replace with PR https://github.com/hpcc-systems/helm-chart/pull/5 in helm-chart project

**Pull request closed without merging**